### ISSUE TITLE
Test scripts

### ISSUE THREAD

**Issue by gusty** [gusty/FSharpPlus#56]
It would be nice to have in the build process a step to compile all scripts from the samples and the doc.
Some scripts are very sensitive to changes and fail to compile after an unnoticed breaking change.

**Comment by wallymathieu** [comment]
Testing the docs and scripts would be helpful, since I noticed some things in 
- [zero](http://gusty.github.io/FSharpPlus/abstraction-monoid.html#zero) is named something else
- [traverse](http://gusty.github.io/FSharpPlus/abstraction-traversable.html#traverse) is missing `Traverse`

**Comment by gusty** [comment]
> zero is named something else

Can you remember where?

If you find more stuff like this feel free to submit a quick PR to fix the docs, if you want.

I was thinking about doing a full findfiles *.fsx and try to compile them with fsc but now I see that that part of the doc would not be easy to test, because it's only a fragment of the "real" code.

**Comment by wallymathieu** [comment]
I think it was due to the editor having wrong search filter that I couldn't find any examples of get_Zero. When I searched for it again, it turned up.

**Comment by wallymathieu** [comment]
The alternative in the [diagram](http://gusty.github.io/FSharpPlus/abstractions.html
) has a zero, while in the docs it's empty.

**Comment by gusty** [comment]
Good catch. I was expecting something like that since it was renamed not long time ago.
I'll fix it and deploy the docs, but before I wan't to add some more sample files.
Keep reporting doc errors in this thread in the meantime.

**Comment by wallymathieu** [comment]
I've added a new pull request to compile the samples at least : https://github.com/gusty/FSharpPlus/pull/62

**Comment by gusty** [comment]
Yes, I saw it. I'm looking into it.
Cool stuff !!!

**Comment by wallymathieu** [comment]
Looks nice 👍  I've not seen it.

**Comment by gusty** [comment]
I just fixed the docs that were not compilable.

That means that now we can do the same processing as you did with the sample files, using the same hack for the moment.

You might have noticed already that there is some overlap between the samples and the docs, since the docs also contains sample code.

I imagine at some point we'll unify them, samples are docs.

**Comment by wallymathieu** [comment]
Yes, and perhaps you could render `expressions`, so that these could be reused and tested?

**Comment by gusty** [comment]
> Yes, and perhaps you could render expressions

Sorry, I didn't get it. What expressions you mean?

**Comment by wallymathieu** [comment]
I was thinking of expressions done by [code quotations](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/code-quotations).

**Comment by wallymathieu** [comment]
But there might be a simpler solution: https://fsprojects.github.io/FSharp.Formatting/evaluation.html

**Comment by gusty** [comment]
Oh I see, you mean quotations. Not sure if I get the whole idea.
What you propose is to include the results of the scripts? Instead of writing ``// val it = 5``?
If so, I guess that's something we should address as well at some point.
And maybe we can include some tests, with the ``hidden``command.

**Comment by wallymathieu** [comment]
Yes, executing tests in order to verify that samples works as expected sounds like a good idea.

**Comment by gusty** [comment]
Now that I moved lot of sample code to the docs, it's more valuable to add compilation to the docs.

Since you're maintainer of FSharp Formatting now, do you know how to configure it to compile the docs?

Also would be nice (and I think it's related to the compilation) to have the feature that emulates intellisense (tooltips) when hovering the mouse over the code in the docs. I know FSharp Formatting support it, since all the samples it its website have it working.

**Comment by wallymathieu** [comment]
I don't think I'm a maintainer of FSharp.Formatting https://github.com/fsprojects/FSharp.Formatting/issues/456#issuecomment-361078737 I talked a little to @matthid about the goals for the projects. Turns out there are more complications than I previously thought. However, it would be nice to get a bit more traction on that project anyway.

I've looked into the [code](https://github.com/fsprojects/FSharp.Formatting) a little, but I don't have grokked it fully.

**Comment by gusty** [comment]
I will see if I can find time to investigate this issue.
Actually I don't remember if it was working when we first incorporated Project Scaffold.
But a quick look into its website reveals that our template is not even close to what they show in their examples.

I just discovered some docs stopped compiling, still the project was building and the docs being released, so I think it's not compiling at all the docs.
